api/wasm/cpp/proxy_wasm_api.h

```diff
@@ -293,60 +280,58 @@ class RootContext : public ContextBase {
 
   // Can be used to validate the configuration (e.g. in the control plane). Returns false if the
   // configuration is invalid.
-  virtual bool validateConfiguration(std::unique_ptr<WasmData> /* configuration */) { return true; }
+  virtual bool validateConfiguration(size_t /* configuration_size */) { return true; }
   // Called once when the VM loads and once when each hook loads and whenever configuration changes.
   // Returns false if the configuration is invalid.
-  virtual bool onConfigure(std::unique_ptr<WasmData> /* configuration */) { return true; }
+  virtual bool onConfigure(size_t /* configuration_size */) { return true; }
   // Called when each hook loads.
-  virtual void onStart(WasmDataPtr /* vm_configuration */) {}
+  virtual void onStart(size_t /* vm_configuration_size */) {}
   // Called when the timer goes off.
   virtual void onTick() {}
   // Called when data arrives on a SharedQueue.
   virtual void onQueueReady(uint32_t /* token */) {}
 
   // Low level HTTP/gRPC interface.
-  virtual void onHttpCallResponse(uint32_t token, std::unique_ptr<WasmData> header_pairs,
-                                  std::unique_ptr<WasmData> body,
-                                  std::unique_ptr<WasmData> trailer_pairs);
+  virtual void onHttpCallResponse(uint32_t token, size_t body_size);
```

Comment:
Because the call to get the body requires the length requested, so it would require an extra call to get the length of the buffer before getting the whole body.